src/Features/LanguageServer/Protocol/Handler/Symbols/DocumentSymbolsHandler.cs

```diff
@@ -29,61 +29,67 @@ public DocumentSymbolsHandler()
         {
         }
 
-        public async Task<object[]> HandleRequestAsync(Solution solution, DocumentSymbolParams request,
-            ClientCapabilities clientCapabilities, string clientName, CancellationToken cancellationToken)
+        public async Task<object[]> HandleRequestAsync(Solution solution,
+            DocumentSymbolParams request,
+            ClientCapabilities clientCapabilities,
+            string? clientName,
+            CancellationToken cancellationToken)
         {
             var document = solution.GetDocumentFromURI(request.TextDocument.Uri, clientName);
             if (document == null)
             {
                 return Array.Empty<SymbolInformation>();
             }
 
-            var symbols = ArrayBuilder<object>.GetInstance();
+            using var _ = ArrayBuilder<object>.GetInstance(out var symbols);
 
             var navBarService = document.Project.LanguageServices.GetRequiredService<INavigationBarItemService>();
             var navBarItems = await navBarService.GetItemsAsync(document, cancellationToken).ConfigureAwait(false);
             if (navBarItems.Count == 0)
             {
-                return symbols.ToArrayAndFree();
+                return symbols.ToArray();
             }
 
             var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
+            if (compilation == null || tree == null)
```

Comment:
Is this for F# support or something else? Or can we not hit this?

Reply:
I believe TS has an export for this... but I'm not sure why as the tree would always be missing and return empty.  will look